source/tutorial/autocomplete-component.md

```diff
@@ -0,0 +1,202 @@
+
+The Handlebars template looks like this:
+
+```app\templates\components\filter-listing.hbs
+City: {{input value=filter key-up=(action 'autoComplete' filter)}} 
+<button {{action 'search'}}>Search</button>
+
+<ul>
+{{#each filteredList as |item|}}
+  <li {{action 'choose' item.city}}>{{item.city}}</li>
+{{/each}}
+</ul>
+```
+It contains an [`{{input}}`](../../templates/input-helpers) helper, that 
+renders as a text field that the user can type in to look for properties 
+in a given city. The `value` property of the `input` will be bound to the 
+`filter` property in our backing JavaScript object. The `key-up` property 
+will be bound to a `autoComplete` action in our backing object, and passes
+ the `filter` property as a parameter.
+
+It also contains a button, whose `action` parameter is bound to the 
+`search` action in our component.
+
+Lastly, it contains an unordered list, that uses the `filteredList` 
+property for data, and displays the `city` property of each item in the 
+list. Clicking the list item will fire the `choose` action, which will 
+populate the `input` field with the name of the `city` in the clicked list
+ item.
+
+Here is what the component's JavaScript looks like:
+
+```app\components\filter-listing.js
+export default Ember.Component.extend({
+  filter: null,
+  filteredList: null,
+  actions: {
+    autoComplete() {
+      this.get('autoComplete')(this.get('filter'));
+    },
+    search() {
+      this.get('search')(this.get('filter'));
+    },
+    choose(city) {
+      this.set('filter',city);
+    }
+  }
+});
+
+```
+There's a property for each of the `filter` and `filteredList`, and 
+actions as described above. What's interesting is that only the `choose` 
+action is defined by the component. The actual logic of each of the 
+`autoComplete` and `search` actions are pulled from the component's 
+properties, which  means that those actions need to be [passed]
+ (../../components/triggering-changes-with-actions/#toc_passing-the-action-to-the-component) 
+ in by the calling object, a pattern known as _closure actions_.
+
+To see how this works, change your `index.hbs` template to look like this:
+
+```app\templates\index.hbs
+<h1>Welcome to Super Rentals</h1>
+
+We hope you find exactly what you're looking for in a place to stay.
+<br /><br />
+{{filter-listing filteredList=filteredList 
+autoComplete=(action 'autoComplete') search=(action 'search')}}
+{{#each model as |rental-unit|}}
+  {{rental-listing rental=rental-unit}}
+{{/each}}
+
+{{#link-to 'about'}}About{{/link-to}}
+{{#link-to 'contact'}}Click here to contact us.{{/link-to}}
+```
+We've added the `filter-listing` component to our `index.hbs` template. We 
+then pass in the functions and properties we want the `filter-listing` 
+component to use, so that the `index` page can define some of how it wants 
+the component to behave, and so the component can have some 
+narrowly-defined access to aspects of the calling scope.
```

Comment:
component to behave, and **so the component can use those specific functions and properties** ?  When we start discussing "calling scope", for folks just learning Ember their eyes are likely to glaze over a bit ;-)